BricksAndHearts.UnitTests/ControllerTests/Landlord/LandLordControllerUnassignedTests.cs

```diff
@@ -0,0 +1,70 @@
+using BricksAndHearts.Database;
+using BricksAndHearts.Services;
+using BricksAndHearts.ViewModels;
+using Castle.Core.Internal;
+using FakeItEasy;
+using FluentAssertions;
+using Microsoft.AspNetCore.Mvc;
+using Xunit;
+
+namespace BricksAndHearts.UnitTests.ControllerTests.Landlord;
+
+public class LandLordControllerUnassignedTests : LandlordControllerTestsBase
+{
+    [Fact]
+    public async void
+        RegisterLandlordWithNoAssignedUser_WhenAttemptedByAdmin_CreatesNewUnassignedLandlord_AndRedirectsToTheirProfile()
+    {
+        // Arrange
+        var adminUser = CreateAdminUser();
+        MakeUserPrincipalInController(adminUser, UnderTest);
+        var returnedLandlord = A.Fake<LandlordDbModel>();
+        var formResultModel = A.Fake<LandlordProfileModel>();
+        formResultModel.Unassigned = true;
+
+        // Act
+        A.CallTo(() => LandlordService.RegisterLandlord(formResultModel))
+            .Returns((ILandlordService.LandlordRegistrationResult.Success, returnedLandlord));
+        A.CallTo(() => MailService.SendMsg(
+            A<string>._, A<string>._, A<string>._, A<string>._
+        )).WithAnyArguments().DoesNothing();
+        var result = await UnderTest.RegisterPost(formResultModel) as RedirectToActionResult;
+
+        // Assert
+        result.Should().BeOfType<RedirectToActionResult>();
+        result.Should().NotBeNull();
+        if (result != null)
+        {
+            result.ActionName.Should().BeEquivalentTo("Profile");
+            result.ControllerName.Should().BeEquivalentTo("Landlord");
+            Assert.False(result.RouteValues.IsNullOrEmpty());
+            result.RouteValues?["id"].Should().Be(returnedLandlord.Id);
+        }
+    }
```

Comment:
Can tidy this up a bit:

* No need to use fakes for simple objects if you don't need to mock anything
* `.DoesNothing();` is the default behaviour of FakeItEasy it seems, so we can omit that setup
* Setting up mocks (`A.CallTo`) normally sits in the Arrange section
* Can be a bit more liberal with the assume not null `!` operator in tests, no need to check separately

```suggestion
    [Fact]
    public async void
        RegisterLandlordWithNoAssignedUser_WhenAttemptedByAdmin_CreatesNewUnassignedLandlord_AndRedirectsToTheirProfile()
    {
        // Arrange
        var adminUser = CreateAdminUser();
        MakeUserPrincipalInController(adminUser, UnderTest);

        var returnedLandlord = new LandlordDbModel
        {
            Id = 5
        };
        var formResultModel = new LandlordProfileModel
        {
            Unassigned = true
        };
        A.CallTo(() => LandlordService.RegisterLandlord(formResultModel))
            .Returns((ILandlordService.LandlordRegistrationResult.Success, returnedLandlord));

        // Act
        var result = await UnderTest.RegisterPost(formResultModel) as RedirectToActionResult;

        // Assert
        result.Should().BeOfType<RedirectToActionResult>();
        result.Should().NotBeNull();
        result!.ActionName.Should().BeEquivalentTo("Profile");
        result.ControllerName.Should().BeEquivalentTo("Landlord");
        result.RouteValues!["id"].Should().Be(returnedLandlord.Id);
    }
```

Reply:
so it's okay to leave the null checks out as they'd cause exceptions only during testing anyway?

Reply:
Yeah. Having `result.Should().NotBeNull();` will give you a slightly nicer error message than a null pointer exception on the next line, but I don't think it's worth the extra code, being able to quickly scan a test and work out what it's doing is helpful.

In non-test code it's good to be explicit about things like that, but it doesn't really matter in tests.